File: c/bubblesort.c
#include<stdio.h>
void bubblesort_pointer(int*,int );
void swap(int*,int*);
int main()
{
    int i,j,*ptr;
    int arr[]={5,4,3,6,8};
    ptr=arr;

    printf("Unsorted array:\n");
    for(i=0;i<5;i++)
    {
        printf("%d ",*(ptr++));
    }
    printf("\n");
    
    bubblesort_pointer(arr,5);
    ptr=arr;
    printf("Sorted array is:\n");
    for ( i = 0; i < 5; i++)
    {
        printf("%d ",*(ptr++));
    }
      
}
void bubblesort_pointer(int arr[],int n)
{
    int *ptr;
    ptr=arr;
    int i,j;
    for (i=0;i<n-1;i++)
    {
        for (j=0;j<n-i-1;j++)
        {
            if(*(ptr+j)>*(ptr+j+1))
            {
                swap(ptr+j,ptr+j+1);
            }
        }
    }
}
void swap(int* a,int* b)
{
    int temp;
    temp=*a;
    *a=*b;
    *b=temp;
    // *a=*a+*b;
    // *b=*a-*b;
    // *a=*a-*b;
}

File: java/BubbleSort.java
package arrayProblems;

public class BubbleSot {

	public static void main(String[] args) {
		
		int arr[] = {1,5,30,10};
		int temp=0;
		
		for(int j=0;j<arr.length;j++) {
			for(int i =0;i<arr.length-1;i++) {
				if(arr[i+1]<arr[i]) {
					temp=arr[i];
					arr[i]=arr[i+1];
					arr[i+1]=temp;
					
					
				}
			}

		}
		
		for(int k=0;k<arr.length;k++) {
			System.out.println(arr[k]+" ");
		}
	}

}
